crypto/evp/digest.c

```diff
@@ -73,6 +74,34 @@ int EVP_MD_CTX_reset(EVP_MD_CTX *ctx)
     return 1;
 }
 
+EVP_MD_CTX *evp_md_ctx_new_ex(EVP_PKEY *pkey, const ASN1_OCTET_STRING *id,
```

Comment:
It appears the only place this is called is via asn1 API's. These sit outside the providers - so this function probably needs a #ifndef FIPS_MODULE around it..

The error you are currently seeing is because EVP_MD_CTX_set_pkey_ctx is not inside the fips module. So putting this function inside the provider means it also tries to find that method inside the provider and fails.

Reply:
I see - thanks for the explanation!
Fixed.